freqtrade/strategy/interface.py

```diff
@@ -134,6 +136,39 @@ def __init__(self, config: dict) -> None:
         self._last_candle_seen_per_pair: Dict[str, datetime] = {}
         super().__init__(config)
 
+        # Gather informative pairs from @informative-decorated methods.
+        self._ft_informative: Dict[
+            Tuple[str, str], Tuple[InformativeData,
+                                   Callable[[Any, DataFrame, dict], DataFrame]]] = {}
+        for attr_name in dir(self.__class__):
+            cls_method = getattr(self.__class__, attr_name)
+            if not callable(cls_method):
+                continue
+            ft_informative = getattr(cls_method, '_ft_informative', [])
+            if not isinstance(ft_informative, list):
+                # Type check is required because mocker would return a mock object that evaluates to
+                # True, confusing this code.
+                continue
+            for informative_data in ft_informative:
+                asset = informative_data.asset
+                timeframe = informative_data.timeframe
+                if asset:
+                    pair = _format_pair_name(self.config, asset)
+                    if (pair, timeframe) in self._ft_informative:
+                        raise OperationalException(f'Informative pair {pair} {timeframe} can not '
+                                                   f'be defined more than once!')
+                    self._ft_informative[(pair, timeframe)] = (informative_data, cls_method)
+                elif self.dp is not None:
+                    for pair in self.dp.current_whitelist():
+                        if (pair, timeframe) in self._ft_informative:
+                            raise OperationalException(f'Informative pair {pair} {timeframe} can '
+                                                       f'not be defined more than once!')
+                        self._ft_informative[(pair, timeframe)] = (informative_data, cls_method)
+
+    def _format_pair(self, pair: str) -> str:
+        return pair.format(stake_currency=self.config['stake_currency'],
+                           stake=self.config['stake_currency']).upper()
```

Comment:
I'd not have this in the code twice (here and in `strategy_helpers.py`...)

Reply:
Another leftover forgotten after i moved it out of the class 😊